collect_app/src/main/java/org/odk/collect/android/application/initialization/upgrade/AppUpgrader.kt

```diff
@@ -0,0 +1,26 @@
+package org.odk.collect.android.application.initialization.upgrade
+
+import org.odk.collect.shared.Settings
+
+class AppUpgrader(private val metaSettings: Settings, private val upgrades: List<Upgrade>) {
+
+    fun upgrade() {
+        upgrades.forEach {
+            val key = it.key()
+
+            if (key == null) {
```

Comment:
Is this case possible?

Reply:
My thinking is we'll want to use the `AppUpgrader` for settings and work manager migrations, which should run on every upgrade (unlike the existing project which only runs once). For some reason, to get the logic covering those different kinds of cases out of my head, so I was sure the structure we had in place would actually work moving forward. Does that make sense?